src/Symfony/Component/PropertyAccess/AccessInfoGuesser.php

```diff
@@ -0,0 +1,259 @@
+                $access[self::ACCESS_NAME] = $getsetter;
+            } elseif ($this->isMethodAccessible($reflClass, '__set', 2)) {
+                $access[self::ACCESS_TYPE] = self::ACCESS_TYPE_PROPERTY;
+                $access[self::ACCESS_NAME] = $property;
+            } elseif ($access[self::ACCESS_HAS_PROPERTY] && $reflClass->getProperty($property)->isPublic()) {
+                $access[self::ACCESS_TYPE] = self::ACCESS_TYPE_PROPERTY;
+                $access[self::ACCESS_NAME] = $property;
+            } elseif ($this->magicCall && $this->isMethodAccessible($reflClass, '__call', 2)) {
+                // we call the getter and hope the __call do the job
+                $access[self::ACCESS_TYPE] = self::ACCESS_TYPE_MAGIC;
+                $access[self::ACCESS_NAME] = $setter;
+            } elseif (null !== $methods = $this->findAdderAndRemover($reflClass, $singulars)) {
+                if (\is_array($value) || $value instanceof \Traversable) {
+                    $access[self::ACCESS_TYPE] = self::ACCESS_TYPE_ADDER_AND_REMOVER;
+                    $access[self::ACCESS_ADDER] = $methods[0];
+                    $access[self::ACCESS_REMOVER] = $methods[1];
+                } else {
+                    $access[self::ACCESS_TYPE] = self::ACCESS_TYPE_NOT_FOUND;
+                    $access[self::ACCESS_NAME] = sprintf(
+                        'The property "%s" in class "%s" can be defined with the methods "%s()" but '.
+                        'the new value must be an array or an instance of \Traversable, '.
+                        '"%s" given.',
+                        $property,
+                        $reflClass->name,
+                        implode('()", "', $methods),
+                        \is_object($value) ? \get_class($value) : \gettype($value)
+                    );
+                }
+            } else {
+                $access[self::ACCESS_TYPE] = self::ACCESS_TYPE_NOT_FOUND;
+                $access[self::ACCESS_NAME] = sprintf(
+                    'Neither the property "%s" nor one of the methods %s"%s()", "%s()", '.
+                    '"__set()" or "__call()" exist and have public access in class "%s".',
+                    $property,
+                    implode('', array_map(function ($singular) {
+                        return '"add'.$singular.'()"/"remove'.$singular.'()", ';
+                    }, $singulars)),
+                    $setter,
+                    $getsetter,
+                    $reflClass->name
+                );
+            }
+        }
+
+        if (isset($item)) {
+            $this->cacheItemPool->save($item->set($access));
+        }
+
+        return $this->writePropertyCache[$key] = $access;
+    }
+
+    /**
+     * Guesses how to read the property value.
+     *
+     * @param string $class
+     * @param string $property
+     *
+     * @return array
+     */
+    public function getReadAccessInfo($class, $property)
```

Comment:
```suggestion
    public function getReadAccessInfo(string $class, string $property): array
```

Typehint can be added everywhere because it's a new class.